tests/test_class_destruction.py

```diff
@@ -23,7 +23,7 @@
     
     struct I1 {
       std::shared_ptr<int> x = std::make_shared<int>(3);
-      virtual ~I1() {}
```

Comment:
For tests, since `{}` is more concise I would lean on keeping it. (or, why would `= default` be better?)
(here and elsewhere)

Reply:
I answered this in the other comment